src/cmap/connection.ts

```diff
@@ -712,10 +711,12 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
       try {
         return await Promise.race([drainEvent, timeout]);
       } catch (error) {
+        let err = error;
         if (TimeoutError.is(error)) {
-          throw new MongoOperationTimeoutError('Timed out at socket write');
+          err = new MongoOperationTimeoutError('Timed out at socket write');
         }
-        throw error;
+        this.cleanup(error);
```

Comment:
Same Q, this cleanup is new for every error but if you saw something related to CSOT shouldn't it be conditional?